include/core/evasion_state.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef INCLUDE_CORE_STATES_HPP_
+#define INCLUDE_CORE_STATES_HPP_
+
+#include <memory>
+#include <mutex>
+#include <functional>
+#include <chrono>
+#include <vector>
+
+#include "core/mission_config.hpp"
+#include "utilities/datatypes.hpp"
+#include "utilities/constants.hpp"
+#include "ticks/ids.hpp"
+#include "ticks/tick.hpp"
+#include "protos/obc.pb.h"
+
+/*
+    State for when plane is actively evading a dynamic obstacle.
+    Plane enters this state after a *dynamic* obstacle is detected. This state
+    will choose a form of active evasion and then return control to normal
+    pathing afterwards.
+    The path planning does not run in real-time, so avoidance behaves by inserting
+    emergency waypoints and then returning to continue the next unhit waypoint.
+*/
+class ObstacleEvasionState {
+    public:
+        // Do not use a default constructor. The evasion state needs to know
+        // the location of the obstacle
+        ObstacleEvasionState() = delete;
```

Comment:
I don't know if you need to delete the default constructor. I think if you define another one it will not generate a default one. But deleting it shouldn't hurt.